oshmem/mca/atomic/basic/atomic_basic_module.c

```diff
@@ -91,6 +92,86 @@ int mca_atomic_basic_finalize(void)
     return OSHMEM_SUCCESS;
 }
 
+static inline
+int mca_atomic_basic_op(void *target,
+                        const void *value,
+                        size_t size,
+                        int pe,
+                        struct oshmem_op_t *op)
+{
+    int rc = OSHMEM_SUCCESS;
+    long long temp_value = 0;
+
+    atomic_basic_lock(pe);
+
+    rc = MCA_SPML_CALL(get(target, size, (void*)&temp_value, pe));
+
+    op->o_func.c_fn((void*) value,
+            (void*) &temp_value,
+            size / op->dt_size);
+
+    if (rc == OSHMEM_SUCCESS) {
+        rc = MCA_SPML_CALL(put(target, size, (void*)&temp_value, pe));
+        shmem_quiet();
+    }
+
+    atomic_basic_unlock(pe);
+
+    return rc;
+}
+
+static inline
+int mca_atomic_basic_fop(void *target,
```

Comment:
looks like no need for "mca_atomic_basic_op", could implement only with "mca_atomic_basic_fop" and pass dummy buffer for "prev"?

Reply:
it will require additional memcpy in case of single_op
ok, done